common/djangoapps/third_party_auth/models.py

```diff
@@ -279,28 +279,34 @@ class ProviderConfig(ConfigurationModel):
             "authentication using the correct link is still possible."
         ),
     )
-    DATA_CONSENT_DISABLED = 'disabled'
-    DATA_CONSENT_OPTIONAL = 'optional'
-    DATA_CONSENT_REQUIRED = 'required'
-    DATA_CONSENT_ENFORCED = 'enforced'
 
-    DATA_CONSENT_STATE_CHOICES = (
-        (DATA_CONSENT_DISABLED, 'Disabled'),
+    DATA_CONSENT_OPTIONAL = 'optional'
+    AT_LOGIN = 'at_login'
+    AT_ENROLLMENT = 'at_enrollment'
+    DATA_SHARING_CONSENT_CHOICES = (
         (DATA_CONSENT_OPTIONAL, 'Optional'),
-        (DATA_CONSENT_REQUIRED, 'Required'),
-        (DATA_CONSENT_ENFORCED, 'Enforced'),
+        (AT_LOGIN, 'At Login'),
+        (AT_ENROLLMENT, 'At Enrollment'),
     )
-    data_sharing_consent = models.CharField(
-        max_length=8,
-        blank=False,
-        choices=DATA_CONSENT_STATE_CHOICES,
-        default=DATA_CONSENT_DISABLED,
+
+    enable_data_sharing_consent = models.BooleanField(
+        default=False,
         help_text=_(
-            "This field is used to determine whether data sharing consent is requested, "
-            "required or enforced of users signing in using this SSO provider. If disabled, consent "
+            "This field is used to determine whether data sharing consent is enabled or "
+            "disabled of users signing in using this SSO provider. If disabled, consent "
             "will not be requested, and course data will not be shared."
         )
     )
+    enforce_data_sharing_consent = models.CharField(
+        max_length=25,
+        blank=False,
+        choices=DATA_SHARING_CONSENT_CHOICES,
+        default=DATA_CONSENT_OPTIONAL,
+        help_text=_(
+            "This field is used to determine the place from where user must consent to data "
```

Comment:
Could use some clarification - "Determines if data sharing consent is optional, if it's required at login, or if it's required when registering for courses" or something to that effect.